drivers/at/at.c

```diff
@@ -99,6 +134,15 @@ int at_expect_bytes(at_dev_t *dev, const char *bytes, uint32_t timeout)
     return res;
 }
 
+int at_wait_bytes(at_dev_t *dev, const char *bytes, uint32_t timeout)
+{
+    int res;
+    do {
+        res = at_expect_bytes(dev, bytes, timeout);
+    } while (res != 0 && res != -ETIMEDOUT);
```

Comment:
The timeout might be reset multiple times.  Is that ok?

Reply:
This is the way timeouts worked the whole time, so I stuck to this behavior.

Reply:
there wasn't a loop before 

before: 
```
at_expect_bytes(dev, bytes, timeout)
````
takes upto timeout time 

this:
```
at_expect_bytes(dev, bytes, timeout); //takes upto timeout time
if (res != 0 && res != -ETIMEDOUT) at_expect_bytes(dev, bytes, timeout) //takes upto timeout time
if (res != 0 && res != -ETIMEDOUT) at_expect_bytes(dev, bytes, timeout) //takes upto timeout time
if (res != 0 && res != -ETIMEDOUT) at_expect_bytes(dev, bytes, timeout) //takes upto timeout time
...

```

Reply:
I'm not fond of this way of handling timeouts either, but this goes all way down to how `isrpipe_read_timeout()` works. On the other hand, the modems usually send data in bursts, so I can hardly imagine a scenario where timeouts massively add up.

Reply:
thanks for clarification